### PR TITLE
Improve logs query by using like

### PR DESCRIPTION
**Pull request by bartv**
# Description

Use like instead of position + fix the name of the lock

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/data/__init__.py [hunk 1]
```diff
@@ -3992,14 +3992,18 @@ def _get_resource_logs_base_query(
         resource_id: m.ResourceIdStr,
         offset: int,
     ) -> Tuple[str, List[object]]:
+        # The query uses a like query to match resource id with a resource_version_id. This means we need to escape the % and _
+        # characters in the query
+        resource_id = resource_id.replace("%", "#%").replace("_", "#_") + "%"
```
Comment (wouterdb) [inline src/inmanta/data/__init__.py:3997]: We probably have to escape itself # as well

File: src/inmanta/data/__init__.py [hunk 2]
```diff
@@ -4787,6 +4794,7 @@ async def get_released_resources(
             db_query = f"""SELECT * FROM ({db_query}) AS matching_records
                         ORDER BY matching_records.{order_by_column} {order}, matching_records.resource_version_id {order}"""
 
+        print(db_query, values)
```
Comment (wouterdb) [inline src/inmanta/data/__init__.py:4797]: This has to go I guess

File: src/inmanta/data/__init__.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/data/__init__.py:3997]: Might be good to test with a resource with a sequence of escaped chars

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: src/inmanta/data/__init__.py (hunk 1 above)
Comment (bartv) [reply src/inmanta/data/__init__.py]: I added an escape for the escape. I did a test on a database and that was indeed required

**Review by sanderr — APPROVED** [review]

**Review by wouterdb — APPROVED** [review]

**Pull request closed without merging**